xgcm/test/datasets.py

```diff
@@ -131,6 +329,12 @@
 
 # include periodicity
 datasets_with_periodicity = {
+    "nonperiodic_sgrid2D": (datasets["sgrid2D"], False),
+    "periodic_sgrid2D": (datasets["sgrid2D"], True),
+    # "xperiodic_sgrid2D": (datasets["sgrid2D"], ["X"]),
+    # "yperiodic_sgrid2D": (datasets["sgrid2D"], ["Y"]),
+    # "sgrid2Dvert": (datasets["sgrid2Dvert"], False),
+    # "sgrid3D": (datasets["sgrid3D"], False),
```

Comment:
Lets actually set up some separate test for the parsing instead of adding to these cases here. I think that makes it much easier to separate e.g. tests of grid operations from the setup. 

I propose to have a matching test module either `test_sgrid.py` or `test_metadata_parser.py` (depending on what we decide above). 

In those tests we can focus purely on the output values of the parser without ever needing to construct a grid object. 

Pseudo example:
```python
def test_sgrid_parser():
    ds = construct_test_dataset(...) # this would use your great test cases from here
    ds_parsed, kwargs_parsed = sgrid.parse_metadata(ds) # syntax might vary here, just an illustration for now
    assert some_attribute_on_dataset == 'expected value'
    assert 'X' in kwargs_parsed['coords'].keys()
    ....
```

We know from other tests that if the correct parameters are passed, xgcm will set up a proper grid. Here we only test that these parameters (for now coords, but could later include metrics?) are properly extracted from the given dataset.

Reply:
Started on this in `xgcm/tests/test_metadata_parsing.py`.
Can discuss more later today but foundations now there.